sources/ClangSharp/Interop/clangsharp.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft and Contributors. All rights reserved. Licensed under the University of Illinois/NCSA Open Source License. See LICENSE.txt in the project root for license information.
+
+// Ported from https://github.com/llvm/llvm-project/tree/llvmorg-9.0.0/clang/include/clang-c
+// Original source is Copyright (c) the LLVM Project and Contributors. Licensed under the Apache License v2.0 with LLVM Exceptions. See NOTICE.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace ClangSharp.Interop
+{
+    public static unsafe partial class clangsharp
+    {
+        private const string libraryPath = "libClangSharp";
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getAttrKind", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_AttrKind Cursor_getAttrKind(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcode", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_BinaryOperatorKind Cursor_getBinaryOpcode(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcodeSpelling", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXString Cursor_getBinaryOpcodeSpelling(CX_BinaryOperatorKind Op);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getDeclKind", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_DeclKind Cursor_getDeclKind(CXCursor C);
```

Comment:
`getDeclKind`, `getAttrKind`, and `getStmtKind` lets you determine the type of unexposed cursors.